plugins/async-node/ios/Tests/AsynNodePluginTests.swift

```diff
@@ -292,6 +292,147 @@ class AsyncNodePluginTests: XCTestCase {
+
+                    expectedMultiNode1Text = textString1
+                    textExpectation.fulfill()
+                }
+
+                if count == 3 {
+                    let newText2 = val
+                        .objectForKeyedSubscript("values")
+                        .objectAtIndexedSubscript(1)
+                        .objectForKeyedSubscript("asset")
+                        .objectForKeyedSubscript("value")
+                    guard let textString2 = newText2?.toString() else { return XCTFail("newText was not a string") }
+
+                    expectedMultiNode2Text = textString2
+                    textExpectation2.fulfill()
+                }
+                
+                if count == 4 {
+
+                let newText3 = val
+                        .objectForKeyedSubscript("values")
+                        .objectAtIndexedSubscript(0)
+                        .objectForKeyedSubscript("asset")
+                        .objectForKeyedSubscript("value")
+                    guard let textString3 = newText3?.toString() else { return XCTFail("newText was not a string") }
+
+                    let newText4 = val
+                        .objectForKeyedSubscript("values")
+                        .objectAtIndexedSubscript(1)
+                        .objectForKeyedSubscript("asset")
+                        .objectForKeyedSubscript("value")
+                    guard let textString4 = newText4?.toString() else { return XCTFail("newText was not a string") }
+
+                    expectedMultiNode3Text = textString3
+                    expectedMultiNode4Text = textString4
+                    textExpectation2.fulfill()
+                }
+            }
+        }
+    })
+
+    player.start(flow: .asyncNodeJson, completion: { _ in})
+
+    wait(for: [handlerExpectation, textExpectation], timeout: 5)
+
+    XCTAssert(count == 2)
+    XCTAssertEqual(expectedMultiNode1Text, "new node from the hook 1")
+
+     var replacementResult = AsyncNodeHandlerType.singleNode(.concrete(context.evaluateScript("""
+                (
+                    {"asset": {"id": "text", "type": "text", "value":"new node from the hook 2"}}
+                )
+            """) ?? JSValue()))
+    
+     args = [plugin.handleAsyncNodeReplacement(replacementResult) ?? JSValue()]
+    
+    let _ = callbackFunction?.call(withArguments: args)
+    
+    XCTAssert(count == 3)
+    XCTAssertEqual(expectedMultiNode2Text, "new node from the hook 2")
+
+
+    wait(for: [textExpectation2], timeout: 5)
+    
+    replacementResult = AsyncNodeHandlerType.emptyNode
+    
+    args = [plugin.handleAsyncNodeReplacement(replacementResult) ?? JSValue()]
+    
+    _ = callbackFunction?.call(withArguments: args)
+
+    XCTAssert(count == 4)
+    // asset that the value at index 0 for the object
+    XCTAssertEqual(expectedMultiNode3Text, "undefined")
```

Comment:
hmm i would expect `textString3` corresponding to `expectedMultiNode3Text`  to have a value of the first node in the original mulitnode of the `action` 

this one 

``` {
               "asset": {
                 "id": "action",
                 "type": "action",
                 "exp": "{{count}} = {{count}} + 1",
                 "label": {
                   "asset": {
                     "id": "test",
                     "type": "text",
                     "value": "test"
                   }
                 }
               }
             },
 ```
             
you can check for the actions label value of `test` but you'll need to change the accessing of the value on line 380

Reply:
updated